src/thread.rs

```diff
@@ -146,6 +159,19 @@ pub fn current() -> Thread {
     }
 }
 
+/// Gets the current thread's clock
+pub fn my_clock() -> VectorClock {
+    ExecutionState::with(|state| {
+        let me = state.current();
+        state.get_clock(me.id()).clone()
+    })
+}
+
+/// Gets the clock for the thread with the given task_id
+pub fn get_clock(task_id: TaskId) -> VectorClock {
+    ExecutionState::with(|state| state.get_clock(task_id).clone())
+}
+
```

Comment:
These aren't specific to threads (they'll work for futures too), and aren't part of the stdlib, so I feel like maybe they belong somewhere else. Bernhard recently added `shuttle::context_switches`, maybe we need to make some new module to contain all this "Shuttle metadata".

Reply:
Since there are only 2 such functions for now, I moved them to `lib.rs` where `context_switches` lives.  We can create a module if we add more to the set?